iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -234,26 +234,44 @@ Viewer.prototype.contents = function() {
     return this._iframes.contents();
 }
 
-
-Viewer.prototype._selectAdjacentTag = function (offset) {
-    var elements = $(".ixbrl-element:not(.ixbrl-continuation)", this._contents);
-    var current = $(".ixbrl-selected:not(.ixbrl-continuation)", this._contents);
-    var next;
-    if (current.length == 1) {
-        next = elements.eq((elements.index(current.first()) + offset) % elements.length);
+// Move by offset (+1 or -1) through the tags in the document in document
+// order.
+//
+// Each element may have one or more tags associated with it, so we need to
+// move through the list of tags associated with the current element before
+// moving to the next/prev element
+//
+Viewer.prototype._selectAdjacentTag = function (offset, currentFact) {
+    const elements = $(".ixbrl-element:not(.ixbrl-continuation)", this._contents);
+    const currentElement = $(".ixbrl-selected:not(.ixbrl-continuation)", this._contents);
+    var nextId;
+
+    if (currentElement.length == 1 && currentFact !== undefined) {
+        // The next ID may be on the current element in the case of double-tagged table cells
+        // Build a window of item IDs consisting of all IDs for the previous,
+        // current and next elements, find the position of the current item ID,
+        // and move by "offset"
+
+        const prev = elements.eq((elements.index(currentElement.first()) - 1) % elements.length);
+        const next = elements.eq((elements.index(currentElement.first()) + 1) % elements.length);
+        const idWindow = prev.data('ivid').concat(currentElement.first().data('ivid'), next.data('ivid'));
+        nextId = idWindow[idWindow.indexOf(currentFact.id) + offset];
```

Comment:
This `currentElement.first()` logic is quite complex, and it took me quite awhile to understand.  If I understand correctly, if we start with `<tx ivid="f1 f2"/><td ivid="f3 f4"/><td ivid="f5 f6/>` and have the middle cell selected, then we want to build an idWindow=[f1,f2,f3,f4,f5,f6] so that we can advance properly.  Is that right?

If so, I wonder if it would be simpler to just populate an `_allIds` list during `_preProcessiXBRL` and remove the `prev`/`next`/`idWindow` complication here and just use that list directly.

Reply:
Yes, that's correct.  

We could improve the readability a bit by renaming `currentElement` to `currentElements`, and then doing `const currentElement = currentElements.first();` but a static list of IDs for the whole document might be cleaner, although it'd need to be one list per document for document sets - assuming we want to keep the current behaviour of next/prev actions wrapping within each document rather than the document set as a whole.

Reply:
I've now found a problem with the current approach, which is that we filter out everything with the `ixbrl-continuation` class, but it's possible for a table element to be both a continuation and a fact, and this filters those out.  I'll implement your suggestion of a list of all IDs, as we can then avoid including continuation IDs at all.